wave_matching/include/wave/matching/ndt.hpp

```diff
@@ -0,0 +1,29 @@
+// Class to wrap PCL ICP matcher
+
+#ifndef WAVE_MATCHING_NDT_HPP
+#define WAVE_MATCHING_NDT_HPP
+
+#include <pcl/point_types.h>
+#include <pcl/point_cloud.h>
+#include <pcl/registration/ndt.h>
+#include "wave/matching/matcher.hpp"
+#include "wave/matching/pcl_common.hpp"
+
+namespace wave {
+
+class NDTMatcher : public Matcher<PCLPointCloud> {
+ public:
+    explicit NDTMatcher(float resolution, const std::string &config_path);
```

Comment:
agreed, and also apparently there are some performance costs to passing primitives by reference.  Top answer here: http://stackoverflow.com/questions/14013139/is-it-counter-productive-to-pass-primitive-types-by-reference